src/Aspire.AppHost.Sdk/SDK/Sdk.in.targets

```diff
@@ -77,7 +77,12 @@
       <_AppHostVersion>%(_AppHostPackageReference.Version)</_AppHostVersion>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(_AppHostVersion)' != '' and $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))">
+    <PropertyGroup Condition="'$(_AppHostVersion)' != ''">
+      <!-- Check if the version is in the format major.minor.patch -->
+      <_IsValidVersion>$([System.Text.RegularExpressions.Regex]::IsMatch('$(_AppHostVersion)', '^\d+\.\d+\.\d+$'))</_IsValidVersion>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(_AppHostVersion)' != '' and '$(_IsValidVersion)' == 'true' and $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))">
```

Comment:
This won't work for customers who have `[8.1.0]` as their version. Are we OK with that?

Also, at what point do we think we can remove this code since it relates to an out of support version.

Reply:
Yeah, thought about that too, but those folks would just get the error like they used to before this fix. As you mention, the main goal we want for this limited fix is to support folks that want to use latest.

Reply:
I'm surprised there isn't a built-in function or task to cover this scenario (semantic version checking). I assume we spoke to the relevant SMEs?

Reply:
@rainersigwald would you mind sharing your perspective here? For context, We have this check somewhere in a targets that checks if a particular version of a packagereference is less than 8.2.0, but this is failing for folks that have the packagereference like `9.*-*`. Is there a way to do this with a built-in function other than MSBuild VersionLessThan?

Reply:
No, we don't have a function for that. AFAIK .NET still doesn't have one we could wrap, though I guess we could call into NuGet like we do for the TF-compat functions. Feel free to file a request at dotnet/msbuild.